NewForce-Cohort-6/tabloidfullstack-cranberry-sauce
# Ticket #26 - Reactivate User Profiles

# Description
This PR is for [Ticket #26](https://trello.com/c/jICdl4PX/26-26-reactivate-a-user-profile), which should allow an admin to select a profile from the list of deactivated user profiles and reactivate it. 

> Note: User type differentiation/User privilege restriction has yet to be added.

# Testing Instructions
1. Commit or stash whatever you're currently working on
2. Git fetch --all
3. Git checkout ac-reactivateUser
4. If you don't see the most recent version of my files, git pull origin ac-reactivateUser
5.  If you need to create the Tabloid database, [use this file to do so](https://github.com/NewForce-Cohort-6/tabloidfullstack-cranberry-sauce/blob/main/SQL/01_Tabloid_Create_DB.sql)
6. If you don't have the Tabloid seed data, [use this file to get it
](https://github.com/NewForce-Cohort-6/tabloidfullstack-cranberry-sauce/blob/main/SQL/02_Tabloid_Seed_Data.sql)
7.  Open a new SQL query and run these:
`ALTER TABLE UserProfile
ADD IsActiveUser bit;`
`UPDATE UserProfile SET IsActiveUser = 1 WHERE Id = 1;`
`UPDATE UserProfile SET IsActiveUser = 1 WHERE Id = 2;`
`UPDATE UserProfile SET IsActiveUser = 1 WHERE Id = 3;`
`UPDATE UserProfile SET IsActiveUser = 0 WHERE Id = 4;`
`UPDATE UserProfile SET IsActiveUser = 0 WHERE Id = 5;`
`UPDATE UserProfile SET IsActiveUser = 0 WHERE Id = 6;`
> Optional: repeat the process for the rest of the profiles in the database by changing the `WHERE Id = #`
8. Execute Tabloid from inside of Visual Studio
![Tabloid](https://user-images.githubusercontent.com/106983427/202474349-d57baf9b-323c-4406-b75a-d8f99a886603.png)
9.  Run NPM Start from inside of the Tabloid client folder
![client](https://user-images.githubusercontent.com/106983427/202473962-42d0b50f-1420-4b34-a522-464dc9fc11d5.png)
10. Login using an email that you know is there, has an active status, and works for you.
11. Once logged-in, select `User Profiles` from the navigation bar
12. You'll be redirected to  `'/profiles'` and you should see a link above the profile list that says `View Deactivated`
13. Click the `View Deactivated` link and you'll be redirected to `'/profiles/deactivated'`, a list of inactive profiles 
14. Click the Activate button on any profile that has one and it should return that profile to the list of active users
15. Click the `View Active` link to return to the list of active users and check that the selected profile is there
16. Use the dev tools or the database to check that the isActive property of the selected profile was set to true
17. Logout of the app and attempt to login as the newly-reactivated user
18. If the feature functions as intended, you will be able to login as that user

# Checklist:
